autotag

```diff
@@ -48,6 +48,19 @@ git fetch --tags
 echo 'Finding most recent tag...'
 
 LAST_TAG=$(git describe --abbrev=0 --tags 2>/dev/null)
+
+# Ensure that tag is not created on same commit as current branch commit
+if [[ ! -z "$LAST_TAG" ]]; then
+  CURRENT_BRANCH=$(git branch --show-current)
+  BRANCH_HASH=$(git rev-list -n 1 $CURRENT_BRANCH)
+  LAST_TAG_HASH=$(git rev-list -n 1 $LAST_TAG)
+
+  if [[ "$BRANCH_HASH" = "$LAST_TAG_HASH" ]]; then
+    echo "Last tag is on the same commit as current branch";
```

Comment:
That's quiet a mouthful. Just say the thing that we are watching out for and alert on it.

```suggestion
    echo "The current commit already has a commit on it. Aborting, to avoid tagging it again."
```

Trailing semi-colons are superfluous here. Got JavaScript / Java on the brain?